state/cleanup.go

```diff
@@ -212,7 +218,41 @@ func (st *State) cleanupMachinesForDyingModel() (err error) {
 	return nil
 }
 
-// cleanupServicesForDyingModel sets all services to Dying, if they are
+// cleanupVolumesForDyingModel sets all persistent volumes to Dying,
+// if they are not already Dying or Dead. It's expected to be used when
+// a model is destroyed.
+func (st *State) cleanupVolumesForDyingModel() (err error) {
+	volumes, err := st.AllVolumes()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	for _, v := range volumes {
+		err := st.DestroyVolume(v.VolumeTag())
+		if err != nil && !IsContainsFilesystem(err) {
+			return errors.Trace(err)
```

Comment:
I don't follow. DestroyVolume returns a typed error if the volume contains a filesystem. If you destroy the filesystem, and the volume is bound to the filesystem, the volume will be destroyed when the filesystem is removed from the model.